cmd/goal/wallet.go

```diff
@@ -134,7 +142,7 @@ var newWalletCmd = &cobra.Command{
 		}
 		reportInfof(infoCreatedWallet, walletName)
 
-		if !recoverWallet {
+		if !recoverWallet && !noPassword {
```

Comment:
I might make `--no-seed` a separate option, assuming you agree with my thought that this should be called `--unencrypted` instead of `--non-interactive`.  I get that you're thinking of the single use case of non-interactive, which ought to imply "no-seed". But if we treat "unencrypted" as the use case, it is separable from "non-interactive".